x/shield/simulation/decoder.go

```diff
@@ -30,7 +30,7 @@ func DecodeStore(cdc *codec.Codec, kvA, kvB tmkv.Pair) string {
 		cdc.MustUnmarshalBinaryLengthPrefixed(kvB.Value, &poolIDB)
 		return fmt.Sprintf("%v\n%v", poolIDA, poolIDB)
 
-	case bytes.Equal(kvA.Key[:1], types.PurchaseKey):
+	case bytes.Equal(kvA.Key[:1], types.PurchaseListKey):
 		var purchaseA, purchaseB types.Purchase
```

Comment:
`types.Purchase` --> `types.PurchaseList`

It seems the decoder has not been enabled. Should insert a line after line 163 in app/sim_test.go to enable it.